### PR TITLE
Adding maven wrapper support 

### PR DESCRIPTION
**Pull request by filipelautert**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
Adding maven wrapper support and using it to build Liquibase instead of maven installed by github actions.

DOCS notes: after merging this PR we will need to fix documentation at https://contribute.liquibase.com/code/get-started/env-setup/ to get rid of step 2 and update step 3

Partially fixes https://github.com/liquibase/liquibase/issues/5334 .

### CODE REVIEW

**Comment by filipelautert** [comment]
Created this new as the previous PR was clashing with a very old branch from pro .

**Comment by sayaliM0412** [comment]
@filipelautert : do we want this to be QA'd ?

**Comment by filipelautert** [comment]
> @filipelautert : do we want this to be QA'd ?

@sayaliM0412 yes please!

**Review by jnewton03 — APPROVED** [review]
🔥

File: .github/workflows/build-branch.yml [hunk 1]
```diff
@@ -85,20 +82,15 @@ jobs:
...
-        with:
-          maven-version: ${{ env.MAVEN_VERSION }}
-
       # Version artifact based off of branch and commit SHA.
       - name: Version Artifact
         run: |
-          mvn versions:set "-DnewVersion=${{ needs.setup.outputs.thisBranchName }}-SNAPSHOT"
-          #mvn versions:set "-DnewVersion=${{ inputs.branchName }}-SNAPSHOT"
+          ./mvnw versions:set "-DnewVersion=${{ needs.setup.outputs.thisBranchName }}-SNAPSHOT"
+          #./mvnw versions:set "-DnewVersion=${{ inputs.branchName }}-SNAPSHOT"
```
Comment (rberezen) [inline .github/workflows/build-branch.yml:89]: should we leave commented line?

**Comment by filipelautert** [comment]
> > > @filipelautert : do we want this to be QA'd ?
> > 
> > 
> > @sayaliM0412 yes please!
> 
> who is going to QA'd it? DevOps, Devs, QAs? cc @filipelautert @sayaliM0412

All of the above?

**Review by rberezen — APPROVED** [review]

**Comment by rberezen** [comment]
did a quick check locally, works good!

**Comment by adrian-velonis1** [comment]
Internal documentation ticket: https://datical.atlassian.net/browse/PD-4220

**Pull request merged**